flink-core/src/main/java/org/apache/flink/api/connector/source/lib/util/RateLimiter.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.connector.source.lib.util;
+
+import java.io.Serializable;
+import java.util.concurrent.CompletionStage;
+
+/** The interface to rate limit execution of methods. */
+public interface RateLimiter extends Serializable {
+
+    /** Returns a future that is completed once another event would not exceed the rate limit. */
```

Comment:
We should clarify that acquire must not be called again until the returned future completes, as otherwise we can go budget.

Reply:
Extended the comment.